spec/models/order_item_spec.rb

```diff
@@ -186,4 +186,57 @@
       end
     end
   end
+
+  describe '#sanitize_parameters' do
+    let(:order_item) { create(:order_item_with_callback, :service_parameters => {'key' => 'val'}) }
+    let(:order_item_sanitized_parameters) { instance_double(Catalog::OrderItemSanitizedParameters, :process => double(:sanitized_parameters => sanitized_parameters)) }
+    before { allow(Catalog::OrderItemSanitizedParameters).to receive(:new).and_return(order_item_sanitized_parameters) }
+
+    context 'when parameters need sanitizing' do
+      let(:sanitized_parameters) { {'key' => '$protected$'} }
+
+      it 'copies original parameters to service_parameters_raw' do
+        expect(order_item.service_parameters).to eq('key' => '$protected$')
+        expect(order_item.service_parameters_raw).to eq('key' => 'val')
+      end
+    end
+
+    context 'when parameters do not need sanitizing' do
+      let(:sanitized_parameters) { {'key' => 'val'} }
+
+      it 'leaves service_parameters_raw empty' do
+        expect(order_item.service_parameters).to eq('key' => 'val')
+        expect(order_item[:service_parameters_raw]).to be_nil
+      end
+    end
+  end
```

Comment:
You're describing the `#sanitize_parameters` method, but I don't see anywhere in here that you're actually calling the method to test it?